ddtrace/tracer/option.go

```diff
@@ -48,6 +53,11 @@ func defaults(c *config) {
 	c.serviceName = filepath.Base(os.Args[0])
 	c.sampler = NewAllSampler()
 	c.agentAddr = defaultAddress
+
+	report := os.Getenv("DD_TRACE_REPORT_HOSTNAME")
+	if b, err := strconv.ParseBool(report); err == nil && b {
+		c.hostname, err = os.Hostname()
```

Comment:
The fact that you reassign `err` here is confusing and makes me wonder "what is it used for?". Can we just turn it into `_`? Also, what happens when the hostname can not be obtained? My feeling is that the user should know, especially if he is setting an env. var, expecting it to have an effect.

Reply:
Changed to actually check the error and log a message.
I'm not sure if it's useful - I expect it to be a very rare case and can't be sure of the quality of the error message the system will provide.
It's also a path that's difficult to write tests for.

Reply:
True. I guess this'll turn into something more elaborate eventually.